test/Microsoft.SqlTools.ServiceLayer.Test.Common/TestConnectionProfileService.cs

```diff
@@ -0,0 +1,206 @@
+        {
+            ConnectionProfile connectionProfile = null;
+
+            //Get the server or profile name for given type to use for database connection
+            TestServerIdentity serverIdentiry = TestServers != null ? TestServers.FirstOrDefault(x => x.ServerType == serverType) : null;
+
+            //Search for the connection info in settings.json
+            if (serverIdentiry == null)
+            {
+                //If not server name found, try to find the connection info for given type
+                connectionProfile = Setting != null && Setting.Connections != null ? Setting.Connections.FirstOrDefault(x => x.ServerType == serverType) : null;
+            }
+            else
+            {
+                //Find the connection info for specific server name or profile name
+                connectionProfile = Setting != null ? Setting.GetConnentProfile(serverIdentiry.ProfileName, serverIdentiry.ServerName) : null;
+            }
+
+            if(connectionProfile == null)
+            {
+                return new ConnectionProfile
+                {
+                    ServerName = "localhost",
+                    Database = "master",
+                    AuthenticationType = AuthenticationType.Integrated
+                };  
+            }
+            return connectionProfile;
+        }
+
+        /// <summary>
+        /// Returns database connection parameters for given server type
+        /// </summary>
+        public async Task<ConnectParams> GetConnectionParametersAsync(TestServerType serverType = TestServerType.OnPrem, string databaseName = null)
+        {
+            ConnectionProfile connectionProfile = GetConnectionProfile(serverType);
+
+            if (connectionProfile != null)
+            {
+                //If the password is empty, get the credential using the service
+                if (string.IsNullOrEmpty(connectionProfile.Password))
+                {
+                    Credential credential = await ReadCredentialAsync(connectionProfile.formatCredentialId());
+                    connectionProfile.Password = credential.Password;
+                }
+
+                ConnectParams connenctParam = CreateConnectParams(connectionProfile, serverType, databaseName);
+               
+                return connenctParam;
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Request a Read Credential for given credential id
+        /// </summary>
+        private async Task<Credential> ReadCredentialAsync(string credentialId)
+        {
+            var credentialParams = new Credential();
+            credentialParams.CredentialId = credentialId;
+
+            ServiceTestDriver driver = Driver;
+            if (driver == null)
+            {
+                TestServiceProvider.InitializeTestServices();
+                return await Task.Factory.StartNew(() => CredentialService.Instance.ReadCredential(credentialParams));
```

Comment:
But the method is beneficial to any caller that wants to call it async, since it's an I/O operation it's a good candidate for this.